strawberry/django/test/client.py

```diff
@@ -0,0 +1,66 @@
+import json
+from typing import Any, Dict, Optional
+
+from typing_extensions import Literal
+
+from strawberry.test import BaseGraphQLTestClient, Response
+
+
+class GraphQLTestClient(BaseGraphQLTestClient):
+    def query(
+        self,
+        query: str,
+        variables: Optional[Dict[str, Any]] = None,
+        headers: Optional[Dict[str, Any]] = None,
+        asserts_errors: Optional[bool] = True,
+        format: Literal["multipart", "json"] = "json",
+        **kwargs,
+    ) -> Response:
+        body: Any = {"query": query}
+
+        if variables:
+            body["variables"] = variables
+
+        if format == "multipart":
+            assert variables is not None
+
+            ref_variable = list(variables.keys())[0]
+            if isinstance(variables.get(ref_variable), dict):
+                ref_variable += f".{list(variables[ref_variable].keys())[0]}"
+
+            if len(kwargs) == 1:
+                file_map = json.dumps({ref_variable: [f"variables.{ref_variable}"]})
+            else:
+                file_map = json.dumps(
+                    {
+                        k: [f"variables.{ref_variable}.{index}"]
+                        for index, k in enumerate(kwargs)
+                    }
+                )
```

Comment:
There's a bit much going on here without comments. Can you add a couple?

Reply:
Added some.
basically, these lines cover the test cases I found in the `test_upload.py` file. 
Thinking now, I have to test more complicated mappings, with folders and single files together...